pkg/plugins/runtime/gateway/generator.go

```diff
@@ -256,6 +256,10 @@ func (g Generator) generateLDS(ctx xds_context.Context, info GatewayListenerInfo
 	for _, hostInfo := range hostInfos {
 		gatewayHosts = append(gatewayHosts, hostInfo.Host)
 	}
+	if info.Listener.CrossMesh {
+		// for cross-mesh, we can only add one listener filter chain as there will not be any (usable) SNI available for filter chain matching
+		gatewayHosts = gatewayHosts[:1]
```

Comment:
Let's add this truncation to https://github.com/kumahq/kuma/blob/09097f213741535e7716703f92c5680188a7095d/pkg/plugins/runtime/gateway/filter_chain_generator.go#L94-L96

instead